pkg/utils/batch_process.go

```diff
@@ -0,0 +1,22 @@
+package utils
+
+func BatchProcess[T interface{}](set []T, batchSize int, f func([]T) error) error {
```

Comment:
interesting to think we hadn't done this yet.  Pretty helpful.

Reply:
I hope that I didn't miss one such thing while browsing the codebase :sweat_smile: